django/core/files/storage.py

```diff
@@ -425,3 +433,56 @@ def _setup(self):
 
 
 default_storage = DefaultStorage()
+
+
+class InvalidStorageError(ImproperlyConfigured):
+    pass
+
+
+class StorageHandler:
```

Comment:
Can we refactor out `BaseHandler` from `BaseConnectionHandler` with a common logic and reuse it here?

Reply:
I don't think we should do it. We could de duplicate the beginning of `__getitem__` method but it would make the code less readable.